app/controllers/api/v1/keywords_controller.rb

```diff
@@ -28,6 +28,15 @@ def create
         end
       end
 
+      def show
+        keyword = current_user.keywords.find_by(id: params[:id])
```

Comment:
How about using `rescue_from` in the ApplicationController?
```ruby
rescue_from ActiveRecord::RecordNotFound do
  render status: :not_found
end
```

In controller you can use `find_by!`, and if the keyword not found it will raise an error. Then the error get handle by `rescue_from`. This can reduce code duplication in the future.
```ruby
def show
  keyword = current_user.keywords.find_by!(id: params[:id])

  render json: KeywordSerializer.new(keyword, include: [:links], params: { show_detail: true })
end
```

This can be refactored even more with `before_action`, so the controller action only contain the actual logic
```ruby
before_action :set_keyword, only: :index

def show
  render json: KeywordSerializer.new(@keyword, include: [:links], params: { show_detail: true })
end

private

def set_keyword
  @keyword = current_user.keywords.find_by!(id: params[:id])
end

```

What do you think? 😄

Reference: https://edgeapi.rubyonrails.org/classes/ActiveSupport/Rescuable/ClassMethods.html

Reply:
Thank you so much for this information. Learn a lot from this.
I will update with `rescue_from` and `before_action`

Reply:
Fixed in [`b92f18b` (#56)](https://github.com/suho/google-search-ruby/pull/56/commits/b92f18b5fb2851ab4eb3b08479e2585871f6a0fc)